README.md

```diff
@@ -497,13 +557,149 @@ New-AzureRmResourceGroup -Name "<name>" -Location "<location>"
+
+```sh
+./certutil ca
+```
+
+and follow the instructions.
+
+### TLS for Elasticsearch HTTP layer
+
+You can secure external access to the cluster with TLS with an external
+loadbalancer or Application Gateway. Configuring TLS for the HTTP layer requires
+`xPackPlugins` and `esHttpSecurity` be set to `Yes`.
+
+#### External load balancer
+
+If you choose `external` as the value for `loadBalancerType`, you must either
+
+* supply a PKCS#12 archive with the `esHttpCertBlob` parameter (and optional 
+passphrase with `esHttpCertPassword`) containing the certs and private key to
+secure the HTTP layer. This certificate will be used by all nodes within the cluster, and
+Kibana will be configured to trust the certificate CA (if CA certs are present within the archive).
+A PKCS#12 archive can be generated using [Elastic's certutil command](https://www.elastic.co/guide/en/elasticsearch/reference/current/certutil.html).
+
+**_or_**
+
+* supply a PKCS#12 archive with the `esHttpCaCertBlob` parameter (and optional 
+passphrase with `esHttpCaCertPassword`) containing the CA which should be used to generate
+a certificate for each node within the cluster to secure the HTTP layer.
+Kibana will be configured to trust the CA and perform hostname verification for presented
+certificates. A PKCS#12 archive can be generated using [Elastic's certutil command](https://www.elastic.co/guide/en/elasticsearch/reference/current/certutil.html)
+
+#### Application Gateway
+
+If you choose `gateway` as the value for `loadBalancerType`, you must
+
+* supply a PKCS#12 archive certificate with the `appGatewayCertBlob` parameter (and optional 
+passphrase with `appGatewayCertPassword`) to secure communication to Application Gateway.
+A PKCS#12 archive can be generated using [Elastic's certutil command](https://www.elastic.co/guide/en/elasticsearch/reference/current/certutil.html)
+
+[Application Gateway](https://azure.microsoft.com/en-au/services/application-gateway/)
+performs SSL offload, so communication from Application Gateway to
+Elasticsearch is not encrypted with TLS by default. TLS to Application Gateway may be sufficient for your needs, but if you would like end-to-end encryption by also configuring TLS for Elasticsearch HTTP layer, you can
+
+* supply a PKCS#12 archive with the `esHttpCertBlob` parameter (and optional 
+passphrase with `esHttpCertPassword`) containing the certs and private key to
+secure the HTTP layer. This certificate will be used by all nodes within the cluster, and
+Kibana will be configured to trust the certificate CA (if CA certs are present within the archive).
+A PKCS#12 archive can be generated using [Elastic's certutil command](https://www.elastic.co/guide/en/elasticsearch/reference/current/certutil.html), and you must
+specify a `--dns <name>` argument with a name that matches that in the `--name <name>` argument.
+
+**_and_**
+
+* supply the public key in CER format for the certificate within the PKCS#12 archive
```

Comment:
`.cer` is just a suffix, there is no `CER` format .`openssl pkcs12 -in http_cert.p12 -out public_key.cer -clcerts -nokeys` will export the certificate in PEM format.   

Also this is not just the public key, but rather the certificate.

Reply:
I agree 😄 I was sticking to the same terminology that Microsoft has used in [their documentation for enabling end-to-end encryption with Application Gateway](https://docs.microsoft.com/en-us/azure/application-gateway/application-gateway-end-to-end-ssl-powershell). I'll update to indicate this is the certificate